File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.py
class Solution(object):
    def findLeastNumOfUniqueInts(self, arr, k):
        mp = list(collections.Counter(arr).values())
        heapq.heapify(mp)
        while k > 0:
            k -= heapq.heappop(mp)
        return len(mp) + (k < 0)

File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.cpp
class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        for(int it : arr) {
            mp.put(it, mp.getOrDefault(it, 0) + 1);
        }

        PriorityQueue<Integer> pq = new PriorityQueue<>(mp.values());
        while(k > 0) {
            k -= pq.poll();
        }

        return k < 0 ? pq.size() + 1 : pq.size();
    }
}